pkg/xds/topology/outbound.go

```diff
@@ -49,26 +56,61 @@ func BuildEndpointMap(
 	for _, externalService := range externalServices {
 		service := externalService.Spec.GetService()
 
-		tlsEnabled := false
-		if externalService.Spec.Networking.Tls != nil {
-			tlsEnabled = externalService.Spec.Networking.Tls.Enabled
+		externalServiceEndpoint, err := buildExternalServiceEndpoint(externalService, mesh.Meta.GetMesh(), loader)
+		if err != nil {
+			log.Info("Unable to create ExternalService endpoint", err)
+			continue
 		}
+		outbound[service] = append(outbound[service], *externalServiceEndpoint)
+	}
+
+	return outbound
+}
 
-		tags := externalService.Spec.GetTags()
+func buildExternalServiceEndpoint(externalService *mesh_core.ExternalServiceResource, mesh string, loader datasource.Loader) (*core_xds.Endpoint, error) {
+	tlsEnabled := false
+	var caCert, clientCert, clientKey []byte = nil, nil, nil
+
+	tags := externalService.Spec.GetTags()
+	if externalService.Spec.Networking.Tls != nil {
+		var err error
+		tlsEnabled = externalService.Spec.Networking.Tls.Enabled
 		if tlsEnabled {
 			tags[`kuma.io/external-service-name`] = externalService.Meta.GetName()
 		}
 
-		outbound[service] = append(outbound[service], core_xds.Endpoint{
-			Target: externalService.Spec.GetHost(),
-			Port:   externalService.Spec.GetPortUInt32(),
-			Tags:   tags,
-			Weight: 1,
-			ExternalService: &core_xds.ExternalService{
-				TLSEnabled: tlsEnabled,
-			},
-		})
+		if externalService.Spec.Networking.Tls.CaCert != nil {
+			caCert, err = loader.Load(context.Background(), mesh, externalService.Spec.Networking.Tls.CaCert)
```

Comment:
ok, so the loader uses ResourceManager instead of cached ReadOnlyResourceManager which will be a strain on storage. Please change
pkg/core/bootstrap/bootstrap.go:73 to use `ReadOnlyResourceManager`.
I checked all the usages, we should be fine with eventually consistency of it.